numpy/random/tests/test_generator_mt19937.py

```diff
@@ -933,6 +933,11 @@ def test_choice_large_sample(self):
         res = hashlib.sha256(actual.view(np.int8)).hexdigest()
         assert_(choice_hash == res)
 
+    def test_choice_array_size_empty_tuple(self):
+        assert_array_equal(random.choice([[1, 2, 3]], size=()), [1, 2, 3])
+        assert_array_equal(random.choice([[1]], size=()), [1])
+        assert_array_equal(random.choice([[1]], size=(), axis=1), [1])
```

Comment:
Just for thoroughness, can you also add tests for `[1, 2, 3]`? I haven't looked in detail, but I'm guessing we don't actually cover that case elsewhere.

Reply:
Done.  I also added the strict parameter to a couple of the tests to ensure that an incorrect scalar array doesn't sneak in there.